src/sql/parts/connection/connectionDialog/connectionDialogService.ts

```diff
@@ -149,16 +151,21 @@ export class ConnectionDialogService implements IConnectionDialogService {
 	}
 
 	private handleOnCancel(params: INewConnectionParams): void {
-		if (params && params.input && params.connectionType === ConnectionType.editor) {
-			this._connectionManagementService.cancelEditorConnection(params.input);
+		if (this.uiController.databaseDropdownExpanded) {
+			this.uiController.closeDatabaseDropdown();
+			this.uiController.databaseDropdownExpanded = false;
```

Comment:
Logically it seems like calling `closeDatabaseDropdown` would cause it to no longer be expanded. Why are they separate?